common/metrics/src/metric_server.rs

```diff
@@ -45,12 +45,12 @@ fn whitelist_metrics(
 // only return the whitelisted metrics
 fn whitelist_json_metrics(
     json_metrics: HashMap<String, String>,
-    whitelist: Vec<String>,
-) -> HashMap<String, String> {
-    let mut whitelist_metrics: HashMap<String, String> = HashMap::new();
-    for (key, val) in json_metrics.iter() {
-        if whitelist.contains(&key) {
-            whitelist_metrics.insert(key.to_string(), val.to_string());
+    whitelist: &'static [&'static str],
+) -> HashMap<&'static str, String> {
+    let mut whitelist_metrics: HashMap<&'static str, String> = HashMap::new();
+    for key in whitelist {
+        if let Some(metric) = json_metrics.get(&key.to_string()) {
```

Comment:
I'm pretty sure you can remove the allocation like this:
```suggestion
        if let Some(metric) = json_metrics.get(&key) {
```

Reply:
@bmwill doesn't seems work :(

```
the trait `std::borrow::Borrow<&&str>` is not implemented for `std::string::String`
```

Reply:
oh sorry it should be:
```suggestion
        if let Some(metric) = json_metrics.get(*key) {
```
idk why its not doing the auto derefing

Reply:
So cool!!